detox/src/devices/drivers/AndroidDriver.js

```diff
@@ -217,12 +240,19 @@ class AndroidDriver extends DeviceDriverBase {
     const spawnFlags = [`-s`, `${deviceId}`, `shell`, `am`, `instrument`, `-w`, `-r`, ...launchArgs, ...additionalLaunchArgs, testRunner];
 
     this.instrumentationProcess = spawnAndLog(this.adb.adbBin, spawnFlags, { detached: false });
+    this.instrumentationProcess.childProcess.stdout.on('data', (raw) => this._extractStackTraceFromInstrumLogs(raw.toString()));
     this.instrumentationProcess.childProcess.on('close', async () => {
       await this._terminateInstrumentation();
       await this.adb.reverseRemove(deviceId, serverPort);
     });
   }
 
+  _extractStackTraceFromInstrumLogs(logsDump) {
+    if (this.instrumentationLogsParser.hasStackTraceLog(logsDump)) {
```

Comment:
Based on the previous comment about chunk data, I would rather suggest a signature like:

```js
this.instrumentationLogsParser.addChunk(logsDumpChunk);
```

so that later you could extract the stack trace:

```js
this.instrumentationLogsParser.getStackTrace();
```

It would be a more future-proof solution if chunk bugs indeed will appear on the horizon.

Reply:
I've done so in my recent commit.